src/lib.rs

```diff
@@ -1114,6 +1112,18 @@ pub mod c_api {
             return set_error(error, ErrorKind::InvalidValue);
         }
 
+        let src_strides = if src_strides.is_null() {
```

Comment:
maybe you can shorten this to something like this?

`let src_strides = (!src_strides.is_null()).then(|| slice::from_raw_parts(src_strides, MAX_NUMBER_OF_PLANES));`